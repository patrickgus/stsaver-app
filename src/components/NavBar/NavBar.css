.NavBar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.NavBar h2 {
  margin: 0;
}

.NavBar h2 a {
  text-decoration: none;
}

.NavBar__not-logged-in {
  display: flex;
  flex-direction: column;
}
