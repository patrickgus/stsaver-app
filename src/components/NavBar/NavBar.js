import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import "./NavBar.css";

class NavBar extends Component {
  handleLogoutClick = () => {
    this.props.history.push("/");
  };

  renderLogoutLink() {
    return (
      <div className="NavBar__logged-in">
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="NavBar__not-logged-in">
        <Link to="/">Login</Link>
        <Link to="/register">Register</Link>
      </div>
    );
  }

  render() {
    return (
      <nav className="NavBar">
        <h3>
          <Link to="/activity">STSaver</Link>
        </h3>
        {this.renderLoginLink()}
      </nav>
    );
  }
}

export default withRouter(NavBar);
