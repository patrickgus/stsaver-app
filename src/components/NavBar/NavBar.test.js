import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import NavBar from "./NavBar";

describe("NavBar component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
