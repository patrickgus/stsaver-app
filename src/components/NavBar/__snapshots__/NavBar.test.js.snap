// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders the UI as expected 1`] = `
<nav
  className="NavBar"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-bars fa-w-14 NavBar__navbar-toggle"
    data-icon="bars"
    data-prefix="fas"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <h2>
    <a
      href="/activity/[object Object]"
      onClick={[Function]}
    >
      S.T.Saver
    </a>
  </h2>
  <ul
    className="NavBar__main-nav"
    id="js-menu"
  >
    <li>
      <a
        className="NavBar__nav-links"
        href="/"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li>
      <a
        className="NavBar__nav-links"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</nav>
`;
